pallets/pallet-did-lookup/src/lib.rs

```diff
@@ -0,0 +1,304 @@
+	pub(crate) type CurrencyOf<T> = <T as Config>::Currency;
+
+	#[derive(Clone, Decode, Debug, Encode, TypeInfo, PartialEq)]
+	pub struct ConnectionRecord<DidIdentifier, Account, Balance> {
+		pub did: DidIdentifier,
+		pub deposit: Deposit<Account, Balance>,
+	}
+
+	pub(crate) type ConnectionRecordOf<T> = ConnectionRecord<DidIdentifierOf<T>, AccountIdOf<T>, BalanceOf<T>>;
+
+	pub const STORAGE_VERSION: StorageVersion = StorageVersion::new(1);
+
+	#[pallet::config]
+	pub trait Config: frame_system::Config {
+		type Event: From<Event<Self>> + IsType<<Self as frame_system::Config>::Event>;
+		type Signature: Verify<Signer = Self::Signer> + Parameter;
+		type Signer: IdentifyAccount<AccountId = AccountIdOf<Self>> + Parameter;
+
+		/// The origin that can associate accounts to itself.
+		type EnsureOrigin: EnsureOrigin<Success = Self::OriginSuccess, <Self as frame_system::Config>::Origin>;
+
+		/// The information that is returned by the origin check.
+		type OriginSuccess: CallSources<AccountIdOf<Self>, DidIdentifierOf<Self>>;
+
+		/// The identifier to which accounts can get associated.
+		type DidIdentifier: Parameter + Default;
+
+		/// The currency that is used to reserve funds for each did.
+		type Currency: Currency<AccountIdOf<Self>> + ReservableCurrency<AccountIdOf<Self>>;
+
+		/// The amount of balance that will be taken for each DID as a deposit
+		/// to incentivise fair use of the on chain storage. The deposit can be
+		/// reclaimed when the DID is deleted.
+		#[pallet::constant]
+		type Deposit: Get<BalanceOf<Self>>;
+
+		/// Weight information for extrinsics in this pallet.
+		type WeightInfo: WeightInfo;
+	}
+
+	#[pallet::pallet]
+	#[pallet::generate_store(pub(super) trait Store)]
+	#[pallet::storage_version(STORAGE_VERSION)]
+	pub struct Pallet<T>(_);
+
+	/// Mapping from account identifiers to DIDs.
+	#[pallet::storage]
+	#[pallet::getter(fn connected_dids)]
+	pub type ConnectedDids<T> = StorageMap<_, Blake2_128Concat, AccountIdOf<T>, ConnectionRecordOf<T>>;
+
+	#[pallet::event]
+	#[pallet::generate_deposit(pub(super) fn deposit_event)]
+	pub enum Event<T: Config> {
+		/// A new association between a DID and an account ID was created.
+		AssociationEstablished(AccountIdOf<T>, DidIdentifierOf<T>),
+
+		/// An association between a DID and an account ID was removed.
+		AssociationRemoved(AccountIdOf<T>, DidIdentifierOf<T>),
+	}
+
+	#[pallet::error]
+	pub enum Error<T> {
+		/// The association does not exist.
+		AssociationNotFound,
+
+		/// The origin was not allowed to manage the association between the DID
+		/// and the account ID.
+		NotAuthorized,
+
+		/// The account has insufficient funds and can't pay the fees or reserve
+		/// the deposit.
+		InsufficientFunds,
+	}
+
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+		/// Associate the given account to the DID that authorized this call.
+		///
+		/// The account has to sign the DID in order to authorize the
+		/// association.
```

Comment:
Is that specific enough? Not sure how the part after `did:kilt:` is called.
ad1e191874ec4002b6c2311c6177ce6520b6d4a4

Reply:
I haven't seen it because the commit is not pushed, but the part after the `did:kilt:` is called method-specific id, but I guess it would be ok to call it identifier.